esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -2570,7 +2580,11 @@ bool CWsWorkunitsEx::onWUQuery(IEspContext &context, IEspWUQueryRequest & req, I
         if (req.getType() && strieq(req.getType(), "archived workunits"))
             doWUQueryFromArchive(context, sashaServerIp.get(), sashaServerPort, *archivedWuCache, awusCacheMinutes, req, resp);
         else if(notEmpty(wuid) && looksLikeAWuid(wuid, 'W'))
+        {
+            if (!validateWsWorkunitAccess(context, wuid, SecAccess_Read))
+                throw makeStringExceptionV(ECLWATCH_ECL_WU_ACCESS_DENIED, "Cannot access the workunit %s.", wuid);
```

Comment:
This sentence is slightly ambiguous.  It could be read "'ws_workunits' cannot access (or open) the workunit".  Rather than the user "is not allowed to access the workunit".

Consider "WorkUnit access denied: %s.".